### PR TITLE
Dicareapi 91/lambda data mapping

### PR DESCRIPTION
**Pull request by natasafrgk**

### CODE REVIEW

File: terraform/api_gateway.tf [hunk 1]
```diff
@@ -124,6 +124,59 @@ resource "aws_api_gateway_integration_response" "mockapi_integration_response" {
...
+  rest_api_id = aws_api_gateway_rest_api.api_gateway.id
+  parent_id   = aws_api_gateway_rest_api.api_gateway.root_resource_id
+  path_part   = "record"
+}
+
+resource "aws_api_gateway_method" "data_mapping_method" {
+  rest_api_id      = aws_api_gateway_rest_api.api_gateway.id
+  resource_id      = aws_api_gateway_resource.data_mapping_resource.id
+  http_method      = "GET"
+  authorization    = "NONE"
```
Comment (natasafrgk) [inline terraform/api_gateway.tf:137]: This annoys SonarCloud. Only picks up this line however and ignores the other path. 
The compliant solution I found was to change it to "AWS_IAM".

**Review by ed-bright — CHANGES_REQUESTED** [review]
Once you've fixed the uri on the api gateway try testing it on postman by grabbing the url and api key from the "build" Github action for the PR

File: terraform/api_gateway.tf [hunk 2]
```diff
@@ -124,6 +124,59 @@ resource "aws_api_gateway_integration_response" "mockapi_integration_response" {
...
+  api_key_required = true
+}
+
+resource "aws_api_gateway_integration" "data_mapping_integration" {
+  rest_api_id             = aws_api_gateway_rest_api.api_gateway.id
+  resource_id             = aws_api_gateway_resource.data_mapping_resource.id
+  http_method             = aws_api_gateway_method.data_mapping_method.http_method
+  integration_http_method = "POST"
+  type                    = "AWS_PROXY"
+  uri                     = aws_lambda_function.lambda.invoke_arn
```
[the remark this answers is not in the document]
Comment (natasafrgk) [reply terraform/api_gateway.tf:147]: Ah, yes! I had a failing build due to changing names too deliberately, so I might have changed it back to `lambda` during the process. I'll amend it.

File: terraform/api_gateway.tf (hunk 1 above)
Comment (natasafrgk) [reply terraform/api_gateway.tf:137]: I agree. Documentation allows for NONE, and IAM would add extra effort and complexity. I wanted to run it past you first.

**Comment by natasafrgk** [comment]
> Once you've fixed the uri on the api gateway try testing it on postman by grabbing the url and api key from the "build" Github action for the PR

When no record was provided :
```
{
    "record": null,
    "message": "failed to get record",
    [...]
}
```

File: terraform/api_gateway.tf [hunk 3]
```diff
@@ -124,6 +124,59 @@ resource "aws_api_gateway_integration_response" "mockapi_integration_response" {
   }
 }
 
+resource "aws_api_gateway_resource" "data_mapping_resource" {
+  rest_api_id = aws_api_gateway_rest_api.api_gateway.id
+  parent_id   = aws_api_gateway_rest_api.api_gateway.root_resource_id
+  path_part   = "record"
```
Comment (ed-bright) [inline terraform/api_gateway.tf:130]: maybe rename this to "map_gpcr"

**Review by ed-bright — CHANGES_REQUESTED** [review]

**Review by ed-bright — APPROVED** [review]

**Pull request merged**